bool      dock_mode       # true = DOCK, false = Undock
---
bool       success         # Did it achieve it?
string     info            # Info